python/cudf/cudf/core/dataframe.py

```diff
@@ -400,13 +400,14 @@ def _setitem_tuple_arg(self, key, value):
                     for i, col in enumerate(columns_df._column_names):
                         self._frame[col].loc[key[0]] = value[:, i]
                 else:
-                    # If the column axis key is 0d, the indexed object
-                    # is a series, the 1d array is assigned to the
-                    # series along the column.
-                    if is_scalar(key[1]):
-                        for i, col in enumerate(columns_df._column_names):
+                    # handle cases where value is 1d object:
+                    # If the value is a range object or if the column axis
+                    # key is 0d, the indexed object is a series; therefore
+                    # the 1d array is assigned to the series along the column.
+                    if isinstance(value, range) or is_scalar(key[1]):
```

Comment:
Sorry about not being clear, I meant that if value is a range, the logic after L403 is run, because `np.ndim(range(N)) == 1`. We don't need to change the logic in this branch.
```suggestion
                    if is_scalar(key[1]):
```